impl/src/main/java/org/ehcache/spi/event/DefaultCacheEventListenerFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ehcache.spi.event;
+
+import org.ehcache.config.CacheConfiguration;
+import org.ehcache.config.event.DefaultCacheEventListenerConfiguration;
+import org.ehcache.config.event.DefaultCacheEventListenerFactoryConfiguration;
+import org.ehcache.event.CacheEventListener;
+import org.ehcache.event.CacheEventListenerFactory;
+import org.ehcache.internal.classes.ClassInstanceProvider;
+
+/**
+ * Created by rism on 3/2/15.
+ */
+public class DefaultCacheEventListenerFactory extends ClassInstanceProvider<CacheEventListener<?, ?>> implements CacheEventListenerFactory{
+
+  public DefaultCacheEventListenerFactory() {
+    super(DefaultCacheEventListenerFactoryConfiguration.class, DefaultCacheEventListenerConfiguration.class);
+  }
+
+  @SuppressWarnings("unchecked")
+  @Override
+  public <K, V> CacheEventListener<K, V> createEventListener(String alias, CacheConfiguration<K, V> cacheConfiguration) {
+    return (CacheEventListener<K, V>) newInstance(alias, cacheConfiguration);
```

Comment:
This is the source of the issue #326 and is linked to a [TODO inside `ClassInstanceProvider`](https://github.com/ehcache/ehcache3/blob/43f3d9cc0808cdabac1214d113a938e3804c486d/impl/src/main/java/org/ehcache/internal/classes/ClassInstanceProvider.java#L49).
I believe figuring this out needs to be part of this PR.